SPFx Library is always included in package

### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [ ] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [X] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- SPFx version : 1.13.1
- Node.js version : 14.15


### Describe the bug / error

SPFx Library is always included in webpart package.

### Steps to reproduce

follow this documentation to create SPFx Lib/SPFx Webpart => https://docs.microsoft.com/en-us/sharepoint/dev/spfx/library-component-tutorial

- Create SPFx Library
- Create SPFx webpart
- Launch npm install for two project
- Create link on SPFx Library (npm link)
- add SPFX library in SPFx webpart package.json
- import and use the lib in code
- create link of lib (npm link mylib)
- bundle/package

I created two project for you with bundle-analyser ^^ =>
[testSPfx_library1.13.zip](https://github.com/SharePoint/sp-dev-docs/files/7624511/testSPfx_library1.13.zip)
[testSPfx1.13.zip](https://github.com/SharePoint/sp-dev-docs/files/7624512/testSPfx1.13.zip)


### Expected behavior

If you test with the projects I uploaded, after bundled, check temp/stats directory and see the result (see screenshot).

![Capture d’écran 2021-11-30 103356](https://user-images.githubusercontent.com/582733/144021920-09a4f54b-271b-4c8e-8dd3-83831ee098f6.png)

I have tested some tips, but SPFx Library is always included in webpart package.

Can someone test? thanks.

ps : on bundle-analyser interface check "Show content of concatenated modules (inaccurate)" and zoom.

after retest, if i'm unlink the spfx library and leave the entry in package.json (`dependencies` node), `gulp bundle --ship` stop with error :

```
Error - [tsc] src/webparts/helloWorld/components/HelloWorld.tsx(5,33): error TS2307: Cannot find module 'test-s-pfx-library-1-13' or its corresponding type declarations.
```
it's a normal behavior, because the lib is no longer referenced.

Detail of my action with unlink :

on SPFx library =>

* `npm instal`
* `gulp build`
* `npm link`

on SPFx webpart =>

* `npm install`
* `npm link test-s-pfx-library-1-13`
=> modifiy code to import/use the lib
* `npm unlink  test-s-pfx-library-1-13`
* check package.json, the spfx lib is referenced in `dependencies` node
* `gulp bundle --ship` => error

but maybe I misunderstand :sweat_smile:

--- a/docs/spfx/library-component-tutorial.md
+++ b/docs/spfx/library-component-tutorial.md
@@ -51,6 +51,7 @@ In this tutorial, you'll create a SharePoint Framework (SPFx) library component 
 
 ## How to consume a third-party SPFx library (for local testing)
 
+1. Run `gulp bundle --ship` from the root directory of library solution. a simple `gulp build` not work.
 1. Run `npm link` from the root directory of library solution. In this case, it would be from the **corporate-library** folder.
 1. Right after creating the symbolic link to your new SPFx library component, don't forget to run at least once `gulp build`. Compile your library component at least once in order to be able to import it as a module into a different project. Remember that the **package.json** says that `"main": "lib/index.js"`, so that location needs to exist before any **import** attempt.
 1. This will create a local npm link to the library with the name, which is provided in the **package.json**.

--- a/docs/spfx/library-component-tutorial.md
+++ b/docs/spfx/library-component-tutorial.md
@@ -55,7 +55,7 @@ In this tutorial, you'll create a SharePoint Framework (SPFx) library component 
 1. Run `npm link` from the root directory of library solution. In this case, it would be from the **corporate-library** folder.
 1. Right after creating the symbolic link to your new SPFx library component, don't forget to run at least once `gulp build`. Compile your library component at least once in order to be able to import it as a module into a different project. Remember that the **package.json** says that `"main": "lib/index.js"`, so that location needs to exist before any **import** attempt.
 1. This will create a local npm link to the library with the name, which is provided in the **package.json**.
-1. Create a web part project in a **separate project folder, so not in the library project folder structure**, following the instructions from [here](https://docs.microsoft.com/sharepoint/dev/spfx/web-parts/get-started/build-a-hello-world-web-part#to-create-a-new-web-part-project). Name your web part **CorporateWebPart**.
+1. Create a web part project in a **separate project folder, so not in the library project folder structure**, following the instructions from [here](/sharepoint/dev/spfx/web-parts/get-started/build-a-hello-world-web-part#to-create-a-new-web-part-project). Name your web part **CorporateWebPart**.
 1. From the root of the new web part folder, run the command `npm link corporate-library`
 1. This will create a symbolic link to that locally built library in to the web part and will make it available to your web part.
 1. Open the web part solution in your preferred editor and navigate to **src/webparts/corporateWebPart/CorporateWebPartWebPart.ts**

--- a/docs/spfx/library-component-tutorial.md
+++ b/docs/spfx/library-component-tutorial.md
@@ -1,7 +1,7 @@
 ---
 title: Build solutions with the library component type in SharePoint Framework
 description: Build solutions with the library component type in SharePoint Framework
-ms.date: 06/19/2020
+ms.date: 12/13/2021
 ms.prod: sharepoint
 ms.localizationpriority: high
 ---

